app/services/exception_safe_response_parser.rb

```diff
@@ -0,0 +1,44 @@
+class ExceptionSafeResponseParser
+  
+  attr_accessor :parsed_response
+
+  def initialize
+    @parsed_response = OpenStruct.new(code: nil, data: nil)
+  end
+
+  def parse(response)
+    parsed_response.code = set_code(response)
+    parsed_response.data = set_data(response)
+    parsed_response
+  end
+
+  private
+
+  def set_data(response)
+    response.is_a?(Exception) ? parse_exception(response) : parse_data(response.data)
+  end
+
+  def parse_data(data)
+    data.is_a?(Hash) ? data : "#{parsed_response.code}#{': ' + data.truncate(30) if data }"
+  end
+
+  def parse_exception(response)
```

Comment:
I suppose this is a little opaque because we're handling the 'non-response' Exception as if it were a real response, inside the response parser... what do you think?  Is this potentially confusing?

Reply:
@aldavidson  ^